jobs/jobs.go

```diff
@@ -83,6 +87,9 @@ func (srv *JobServer) SetJobSuccess(job *model.Job) *model.AppError {
 	if _, err := srv.Store.Job().UpdateStatus(job.Id, model.JOB_STATUS_SUCCESS); err != nil {
 		return model.NewAppError("SetJobSuccess", "app.job.update.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
+
```

Comment:
We'll also need to decrement for canceled, error, and requestCancellation

Reply:
Ah, yes, thank you! I had meant to ask about these and address them preemptively because I wasn't confident about the state changes, but I've pushed a change now that amounts to:
* When `SetJobError` transitions to `JOB_STATUS_ERROR` (either unilaterally, or successfully after an optimistic update), we decrement.
* On `SetJobCanceled`, we decrement.
* If `RequestCancellation` successfully transitions to `JOB_STATUS_CANCELED`, we decrement.

In all other cases, in particular when we only transition to `JOB_STATUS_CANCEL_REQUESTED`, we don't decrement, but wait for the end state transition from a `SetJobCanceled` or a `SetJobError`. Let me know if my assumptions are valid here :)